src/goMain.ts

```diff
@@ -87,14 +87,24 @@ export function activate(ctx: vscode.ExtensionContext): void {
 		ctx.globalState.update('goroot', currentGoroot);
 
 		offerToInstallTools();
-		let langServerAvailable = checkLanguageServer();
-		if (langServerAvailable) {
-			let langServerFlags: string[] = vscode.workspace.getConfiguration('go')['languageServerFlags'] || [];
+		let supportedLangServerFlags = checkLanguageServer();
+		if (supportedLangServerFlags) {
+			let configuredLangServerFlags: string[] = vscode.workspace.getConfiguration('go')['languageServerFlags'] || [];
+			configuredLangServerFlags = [
+				...configuredLangServerFlags,
+				'-gocodecompletion',
+				`-func-snippet-enabled=${vscode.workspace.getConfiguration('go')['useCodeSnippetsOnFunctionSuggest']}`
```

Comment:
The 2 flags for completion should only be added if they are present in `supportedLangServerFlags`, else `go-langserver` will fail